main.c

```diff
@@ -128,51 +128,55 @@ static void apply_profile(struct kanshi_state *state,
 	zwlr_output_configuration_v1_add_listener(config, &config_listener, pending);
 
 	size_t i = 0;
+	struct kanshi_head *head;
 	struct kanshi_profile_output *profile_output;
-	wl_list_for_each(profile_output, &profile->outputs, link) {
-		struct kanshi_head *head = matches[i];
-
-		bool enabled = head->enabled;
-		if (profile_output->fields & KANSHI_OUTPUT_ENABLED) {
-			enabled = profile_output->enabled;
-		}
+	while ((head = matches[i]) != NULL) {
```

Comment:
The `matches` array isn't NULL-terminated, so we need to use a classic `for` loop here